mmocr/datasets/pipelines/transform_wrappers.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import inspect
+import random
+
+import mmcv
+import numpy as np
+import torchvision.transforms as transforms
+from mmcv.utils import build_from_cfg
+from mmdet.datasets.builder import PIPELINES
+from mmdet.datasets.pipelines import Compose
+from PIL import Image
+
+
+@PIPELINES.register_module()
+class OneOf:
+    """Randomly select and apply one of the transforms, each with the equal
+    chance.
+
+    Warning:
+        Different from albumentations, this wrapper only runs the selected
+        transform, but doesn't guarantee the transform can always be applied to
+        the input if the transform comes with a probability to run.
+
+    Args:
+        transforms (list[dict]): Candidate transforms to be applied.
+    """
+
+    def __init__(self, transforms):
+        assert isinstance(transforms, list) or isinstance(transforms, tuple)
+        assert len(transforms) > 0
+        self.transforms = []
+        for t in transforms:
+            if isinstance(t, dict):
+                self.transforms.append(build_from_cfg(t, PIPELINES))
+            elif callable(t):
+                self.transforms.append(t)
+            else:
+                raise TypeError('transform must be callable or a dict')
+
+    def __call__(self, results):
+        return random.choice(self.transforms)(results)
+
+    def __repr__(self):
+        repr_str = self.__class__.__name__
+        repr_str += f'(transforms={self.transforms})'
+        return repr_str
+
+
+@PIPELINES.register_module()
+class RunWithProb:
+    """Run a transform or a sequence of transforms with probability p.
+
+    Args:
+        transforms (list[dict]): Transform(s) to be applied.
+        p (float): Probability of running transform(s).
+    """
+
+    def __init__(self, transforms, p):
+        assert type(p) is float
+        self.transforms = Compose(transforms)
```

Comment:
Need to judge transforms' type and if only one transform which is callable, need to convert it type to list